### PR TITLE
Should fix parser.

### PR DESCRIPTION
**Pull request by WildCard65**

### CODE REVIEW

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -6577,7 +6577,7 @@ stock GetBossIndex(client)
 	return -1;
 }
 
-stock Operate(Handle:sumArray, &bracket, Float:value, Handle:_operator)
+stock Operate(Handle:sumArray, bracket, Float:value, Handle:_operator)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: So we never modify bracket anymore?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -6631,27 +6630,29 @@ stock ParseFormula(boss, const String:key[], const String:defaultFormula[], defa
 	KvRewind(BossKV[Special[boss]]);
 	KvGetString(BossKV[Special[boss]], "name", bossName, sizeof(bossName), "=Failed name=");
 	KvGetString(BossKV[Special[boss]], key, formula, sizeof(formula), defaultFormula);
-	ReplaceString(formula, sizeof(formula), " ", "");  //Get rid of spaces
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :+1:

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 1 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I made this to mirror my Eval Stock PR for SM

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 3]
```diff
@@ -6631,27 +6630,29 @@ stock ParseFormula(boss, const String:key[], const String:defaultFormula[], defa
...
-
 	new bracket;  //Each bracket denotes a separate sum (within parentheses).  At the end, they're all added together to achieve the actual sum.
-	new Handle:sumArray=CreateArray(), Handle:_operator=CreateArray();
+	new Handle:sumArray=CreateArray(.startsize=1), Handle:_operator=CreateArray(.startsize=1);
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: `CreateArray(_, 1)`

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 2 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I made the parser ignore whitespace(by handling it)

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 3 above)
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: And what does this fix?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 4]
```diff
@@ -6631,27 +6630,29 @@ stock ParseFormula(boss, const String:key[], const String:defaultFormula[], defa
...
 	decl String:character[2], String:value[1024];
 	for(new i; i<=strlen(formula); i++)
 	{
 		character[0]=formula[i];  //Find out what the next char in the formula is
 		switch(character[0])
 		{
+			case ' ': //Ignore whitespace.
+				continue;
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Brackets (and I'd prefer two spaces before the comment)

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 5]
```diff
@@ -6662,7 +6663,6 @@ stock ParseFormula(boss, const String:key[], const String:defaultFormula[], defa
 					SetArrayCell(sumArray, 0, 0.0);
 					break;
 				}
-
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :frowning: Add this back please?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 6]
```diff
@@ -204,7 +204,7 @@ new changeGamemode;
 
 enum Operators
 {
-	Operator_None=-1,
+	Operator_None=0,
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :+1:

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 3 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: It fixes the odd chance I had when I made the eval Stock for SM where I ran into invalid index 0

**Pull request merged**